utils/stats.py

```diff
@@ -128,5 +135,84 @@ async def update_stats(user: User, now: datetime, daily_reset: bool = False, wee
 
     await user.save_changes()
 
+
+    # For the user in all servers, update their streak and milestone roles
+    result = await Server.find_all("users.id" == user.id).to_list()
+
+    for server_id in result:
+        await give_user_streak_role(user, server_id.id, user.scores.streak)
+        # logger.info("file: utils/message_scheduler.py ~ send_daily_question_and_update_stats ~ updated streak role for user %s", user.id)
+        await give_user_milestone_role(user, server_id.id, user.submissions.total_score)
+
     logger.info(
         'file: cogs/stats.py ~ update_stats ~ user stats updated successfully: %s', leetcode_username)
+    
+
+async def give_user_streak_role(user: discord.User, guild_id: int, streak: int) -> None:
+    logger.info(f"file: utils/stats.py ~ give_user_streak_role ~ run - {user.id} - {streak}")
+    guild = client.get_guild(guild_id)
+
+    if not guild:
+        return
+    
+    discord_user = guild.get_member(user.id)
+
+    if not discord_user:
+        return
+    
+    role_to_assign = None
+    
+    for role_streak, (role_name, role_color) in STREAK_ROLES.items():
+        if streak >= role_streak:
+            role_to_assign = discord.utils.get(guild.roles, name=role_name)
+    
+    # Remove all other streak roles
+    for role_streak, _ in STREAK_ROLES.items():
+        role_name, _ = STREAK_ROLES.get(role_streak)
+        role = discord.utils.get(guild.roles, name=role_name)
+        if (role and role in discord_user.roles and role != role_to_assign) or role_to_assign is None:
+            await discord_user.remove_roles(role)
+            
+    if role_to_assign:
+        # Give the user the appropriate streak role
+        await discord_user.add_roles(role_to_assign)
+        logger.info(f"Assigned {role_to_assign.name} role to {discord_user.display_name}")
+    else:
+        logger.warning("No suitable streak role found.")
```

Comment:
Realistically this could be abstracted, though to be honest, I'm not too bothered to do this 😄

Reply:
Yh definitely should be combined into one function. lol, fair.

Reply:
actually, nvm, the logic is different ofc. Would be a hassle to merge